### DISCUSSION TITLE
Clarifications regarding YOLOv8 licensing

### DISCUSSION THREAD

**Discussion by bdawton** [keras-team/keras-cv#2032] [Q&A]
First of all, thank you for the work you've put into this exciting and helpful project!

I have a question regarding the commercial use of YOLOv8. It seems that the keras-cv implementation is covered by the Apache V2.0  license whilst the Ultralytics version seems to use the much more restrictive AGPL3 license.

I was wondering if the keras-cv YOLOv8 was developed from the ground up? If so, does this mean it's not bound by the stricter licensing associated with the Ultralytics version?

I'm mainly trying to understand if using the keras-cv YOLOv8 for commercial purposes would be okay. Any insight or guidance you can provide on this would be helpful.

Thanks for your time and consideration.

**Comment by innat** [comment]
cc. @ianstenbit

**Comment by aamirshah86** [comment]
Waiting for the answer.

**Comment by Mattral** [comment]
I guess that the licensing of a software project is a crucial factor to consider when using it for commercial purposes. The difference in licenses between the Keras-CV YOLOv8 implementation and the Ultralytics YOLOv8 implementation could indeed have implications for commercial use

Keras-CV YOLOv8 (Apache License 2.0): The Apache License 2.0 is generally considered more permissive and business-friendly compared to the AGPL-3.0 license. It allows you to use, modify, and distribute the software for commercial purposes, as long as you follow the license terms. This license emphasizes the freedom to use the software without extensive restrictions on usage.

Ultralytics YOLOv8 (AGPL-3.0): The GNU Affero General Public License (AGPL-3.0) is a more restrictive license, especially for commercial use. Under the AGPL-3.0, if you use the software in a way that provides a service over a network, you are typically required to share the modified source code of the software. This can make it less suitable for commercial projects that involve proprietary or closed-source components.

As such, (only in case) if Keras-CV YOLOv8 was developed independently and not derived from the Ultralytics version, and it is indeed licensed under the Apache License 2.0, it should be more permissive for commercial use.
To ensure full compliance with licensing and to get the most accurate guidance, I recommend reviewing the specific license terms of the Keras-CV YOLOv8 repository, verifying its origin.

**Reply by innat** [reply]
> As such, (only in case) if Keras-CV YOLOv8 was developed independently and not derived from the Ultralytics version, and it is indeed licensed under the Apache License 2.0, it should be more permissive for commercial use.

That is the concern, many parts are derived form ultralytics, not to mention they developed it.

**Reply by casperroo** [reply]
The question is: is it lawfully licensed under the The Apache License 2.0?
Looking at the source code of the two implementations it is hard to call it "derived from ultralytics". @innat: could you  show some example?

If the implementation is not derived, the next question is: what is the license of the schema/idea/algorithm of the model? I cannot find any white papers on that, all I can find is this:
[Brief summary of YOLOv8 model structure](https://github.com/ultralytics/ultralytics/issues/189)
So as the idea/algorithm was not patented or anything like that, isn't it true that anyone can implement it? If that is the case it seems that the AGPL license applies only to the ultralytics's implementation itself.

**Reply by 4-alok** [reply]
What are the differences between Keras-CV YOLOv8 and Ultralytics YOLOv8 implementations, apart from licensing and ease of use?

**Comment by Quakumei** [comment]
Any updates on that matter?
